python/GafferUI/SpreadsheetUI/_PlugTableView.py

```diff
@@ -217,6 +222,44 @@ def getVisibleSection( self ) :
 
 		return self.__visibleSection
 
+	def setRowFilter( self, pattern ) :
+
+		self.__rowFilterPattern = pattern
+		self.__rowsToFilter = set()
+
+		model = self._qtWidget().model()
+		with self.ancestor( GafferUI.PlugValueWidget ).getContext() :
+			for i in range( 0, model.rowCount() ) :
+				plug = model.plugForIndex( model.index( i, 0 ) )
+				if plug is not None :
+					rowPlug = plug.ancestor( Gaffer.Spreadsheet.RowPlug )
+					rowName = rowPlug["name"].getValue()
+					self.__rowsToFilter.add( ( rowPlug, rowName ) )
+
+		self.__applyRowFilter()
+
+	def getRowFilter( self ) :
+
+		return self.__rowFilterPattern
+
+	def __applyRowFilter( self ) :
+
+		model = self._qtWidget().model()
+		with self.ancestor( GafferUI.PlugValueWidget ).getContext() :
+			for i in range( 0, model.rowCount() ) :
+				visible = True
+				if self.__rowFilterPattern :
+					plug = model.plugForIndex( model.index( i, 0 ) )
+					if plug is not None :
+						rowPlug = plug.ancestor( Gaffer.Spreadsheet.RowPlug )
+						rowName = rowPlug["name"].getValue()
+						if ( rowPlug, rowName ) in self.__rowsToFilter :
+							visible = IECore.StringAlgo.matchMultiple( rowName, self.__rowFilterPattern )
```

Comment:
There's something I don't understand here. If we only ever call `matchMultiple()` when `rowName` is the same as it was in `setRowFilter()`, then why don't we just call `matchMultiple()` in `setRowFilter()`? Unless I'm missing something, that would be equivalent to my earlier suggestion of just creating a set of `self.__rowPlugsToHide` in `setRowFilter()` and then applying that in `__applyRowFilter()`. That would be quicker to execute and I think it would also be easier to understand (because name was only dealt with once in the first function).

Reply:
Ah, I see! Yes, that would be more straightforward indeed. I've gone ahead and adjusted that behaviour before merging.